js/index.js

```diff
@@ -135,9 +134,39 @@ form.addEventListener('submit', (event) => {
   if (!emailRegExp.test(email.value.trim())) {
     form.submit();
   } else {
-
     event.preventDefault();
     //emailClass.classList.toggle('error');
-    errMessageZone.innerText = 'Please enter an entirely lower case email, please';
+    errMessageZone.innerText =
+      'Please enter an entirely lower case email, please';
   }
 });
+
+// preserving Data
+
+let data = {};
```

Comment:
- It would be better if you selected the input fields first and added them to your data object. It can look similar to this. The switch statement you wrote changes all the input values when data is re-entered and returns undefined. 👍 
```
  const fullName = document.getElementById('fullname');
  const email = document.getElementById('email');
  const message = document.getElementById('message-input');

  document.addEventListener('change', () => {

    const data = {
      fullName: fullName.value,
      message: message.value,
      email: email.value,
    };
    localStorage.setItem('data', JSON.stringify(data));
  });

```